<launch>
  number of moving objects
  <arg name="obj_num" value="50" />
  <arg name="drone_id" value="1"/>

  <arg name="map_size_x" value="15.0"/>
  <arg name="map_size_y" value="10.0"/>
  <arg name="map_size_z" value=" 5.0"/>
  <arg name="init_x" value="-9.0"/>
  <arg name="init_y" value="  0.0"/>
  <arg name="init_z" value="  1.0"/>
  <arg name="number_sample" value="1000"/>
  <arg name="odom_topic" value="visual_slam/odom" />


  <node name="prm_planner" pkg="planner" type="planner_node" output="screen" >
    <param name="number_sample" value="$(arg number_sample)" />
    <remap from="/grid_map/x_size" to="$(arg map_size_x)" />
    <remap from="/grid_map/y_size" to="$(arg map_size_x)" />
    <remap from="/grid_map/z_size" to="$(arg map_size_x)" />
    <param name="x_size" value="$(arg map_size_x)"/>
    <param name="y_size" value="$(arg map_size_y)"/>
    <param name="z_size" value="$(arg map_size_z)"/>
    <remap from="~cloud_in" to="/grid_map/occupancy_inflate" />
    <remap from="~odometry" to="$(arg odom_topic)" />
    <param name="resolution" value="0.05" />
    <param name="obstacle_inflation" value="0.2" />
  </node>


  <node pkg="traj_opt" type="traj_opt_node" name="traj_opt_node" output="screen">
    <param name="speed" value="1.0"/>
    <remap from="~waypoint" to="/prm_planner/raw_path"/>
    <remap from="~map" to="/prm_planner/occupancy_inflate" />
    <remap from="/grid_map/x_size" to="$(arg map_size_x)" />
    <remap from="/grid_map/y_size" to="$(arg map_size_y)" />
    <remap from="/grid_map/z_size" to="$(arg map_size_z)" />
    <remap from="~position_command" to="/position_cmd"/>
    <remap from="~odom" to="$(arg odom_topic)"/>
  </node>


</launch>